### ISSUE TITLE
forge.util.text.utf16.encode

### ISSUE THREAD

**Issue by dimitrirostavo** [digitalbazaar/forge#319]
I think there is a bug in forge.util.text.utf16.encode method. 

When output is not provided it should return view and not "out" which is what it does now.

```javascript
util.text.utf16.encode = function(str, output, offset) {
  var out = output;
  if(!out) {
    out = new Uint8Array(str.length);
  }
  var view = new Uint16Array(out);
  offset = offset || 0;
  var j = offset;
  var k = offset;
  for(var i = 0; i < str.length; ++i) {
    view[k++] = str.charCodeAt(i);
    j += 2;
  }
 // return output ? (j - offset) : out;
 return output ? (j - offset) : view;
};

**Comment by dimitrirostavo** [comment]
Please let me know if I am right ? Thanks. out is empty without this fix.

**Comment by davidlehn** [comment]
Can you please provide a minimal runnable example that shows the problem?  Thanks.

**Comment by dimitrirostavo** [comment]
Sorry took so long. 

Sure I can provide sample code. 
If I call into 

I am trying to call this function like this 

var string = "foo"; 

`forge.util.text.utf16.encode(string);`

Now according to the docs I should get back a buffer that has the UTF16 LE representation of it. But I always get back null. Because out is returned internally and never updated.

**Comment by dlongley** [comment]
@dimitrirostavo,

We haven't gotten around to creating a stable API for encoding/decoding UTF-8/UTF-16 in forge 0.6.x ... it was likely going to be left to forge 0.7.x. The current API is experimental and not documented in the examples: (https://github.com/digitalbazaar/forge/blob/master/js/util.js#L1786-L1791).

That being said, I pushed a change to fix the experimental UTF-16 stuff and pushed a couple tests. So, this should be working now. You'll find the fix in forge 0.6.38.
